### PR TITLE
init work on latest metrics

### PR DESCRIPTION
**Pull request by vladikoff**
- [x] Add debounce
- [x] Add a generic POST request to GA, remove the current metrics library.
- [ ] Check the missing tags
- [ ] Check with web-extension team if we can detect close events
- [x] Add DNT (Do Not Track)

Re: GA

> [10:03:11]  <@chuck>	vladikoff: just the GA measurement protocol, which is a simple POST to GA, with the data URL-encoded in the body. More info here: https://developers.google.com/analytics/devguides/collection/protocol/v1/devguide
[10:03:24]  <@chuck>	+ here, which should be linked in the doc in your repo: https://github.com/mozilla/testpilot/blob/master/docs/experiments/ga.md

### CODE REVIEW

**Comment by Natim** [comment]
This first version should work properly.

File: src/background.js [hunk 1]
```diff
@@ -12,26 +14,88 @@ browser.storage.local.get('UID').then((data) => {
...
+
+      if (navigator.doNotTrack === 'unspecified') {
+        return fetch('https://www.google-analytics.com/collect', request);
+      } else {
+        console.log('Do not track activated, not sending anything.');
+        return Promise.resolve();
+      }
+    };
+    clearTimeout(timeouts[event]);
+    timeouts[event] = setTimeout(later, 20000);
```
Comment (Natim) [inline src/background.js]: Refs https://github.com/mozilla/notes/pull/97#discussion_r123174758

**Comment by Natim** [comment]
Can we figure out a way not to send a changed even when the pad opens;

File: src/background.js [hunk 2]
```diff
@@ -12,26 +14,94 @@ browser.storage.local.get('UID').then((data) => {
...
+        cd1: context.syncEnabled,
+        cd2: context.usesSize,
+        cd3: context.usesBold,
+        cd4: context.usesItalics,
+        cd5: context.usesStrikethrough,
+        cd6: context.usesList,
+        cd7: undefined, // Firefox UI used to open, close notepad
+        cd8: undefined, // reason editing session ended
+      };
+      console.log(event, context, gaEvent);
```
Comment (vladikoff) [inline src/background.js]: remove the console log?

**Comment by vladikoff** [comment]
@Natim can you remove the metrics lib from `https://github.com/mozilla/notes/blob/master/package.json#L11` ?

File: src/background.js [hunk 3]
```diff
@@ -12,26 +14,94 @@ browser.storage.local.get('UID').then((data) => {
...
+      }
+      formBody = formBody.join('&');
+
+      const request = {
+        method: 'POST',
+        headers: {'Content-Type': 'application/x-www-form-urlencoded'},
+        body: formBody
+      };
+
+      if (navigator.doNotTrack === 'unspecified') {
```
Comment (vladikoff) [inline src/background.js]: 👍

**Comment by vladikoff** [comment]
Works 
image

**Pull request merged**